*Dataset que contiene información acerca de Tips (consejos) escritos por el usuario. Los tips son más cortas que las reseñas y tienden a dar sugerencias rápidas.*

In [1]:
import pandas as pd
import Utilidades as ut

In [2]:
df = pd.read_csv('../Datos/Archivos csv/tip.csv')
df.head(2)

user_id             business_id  \
0  AGNUgVwnZUey3gcPCJ76iw  3uLgwr0qeCNMjKenHJwPGQ   
1  NBN4MgHP9D3cw--SnauTkA  QoezRbYQncpRqyrLH6Iqjg   

                                                text                 date  \
0                     Avengers time with the ladies.  2012-05-18 02:17:21   
1  They have lots of good deserts and tasty cuban...  2013-02-05 18:35:10   

   compliment_count  
0                 0  
1                 0

Verificamos los registros duplicados

In [3]:
ut.registros_duplicados(df)

user_id             business_id  \
1644    SNZJNz1WEIMECEk-TAYADQ  6Gx45ltNNZvjzVBO6ZgVOg   
11367   00Cz_vdInMHpTRjqbWjK5Q  ncacMQ9n_dSM1cR3c1vTQw   
21583   Tf63LdfubjlFm974OrdTXg  CyiN_owscPJNvlwul0XJPg   
29901   00Cz_vdInMHpTRjqbWjK5Q  ncacMQ9n_dSM1cR3c1vTQw   
32192   y_PMpgAxTWUIYAmzfFc6VA  Bw0cpnfI-Y9QmjWzqS-oFg   
...                        ...                     ...   
820290  wYo3aBVj-bRPT7E4RSj5Kg  W5SNps2JaT_RozLAl_TN1Q   
869840  TEjijSowDwJM4vCL-zn4ew  YPHDzg1h-PkaxfoppT-iJg   
873070  TEjijSowDwJM4vCL-zn4ew  YPHDzg1h-PkaxfoppT-iJg   
900543  XJcb75c9OY0xX8spM0TQuw  UlGS2NxZSiph33E3nfovRw   
903817  XJcb75c9OY0xX8spM0TQuw  UlGS2NxZSiph33E3nfovRw   

                                                     text  \
1644                               Great food, fun place!   
11367                                         Miss saigon   
21583                                                  LM   
29901                                         Miss saigon   
32192                      Late dinner w @steinerrecliner   
...                                                   ...   
820290  Worker there are the best at serving me quick ...   
869840                                         bleed blue   
873070                                         bleed blue   
900543  Whiskey Kitchen is the perfect place for bar f...   
903817  Whiskey Kitchen is the perfect place for bar f...   

                       date  compliment_count  
1644    2017-10-10 20:13:08                 0  
11367   2010-12-07 01:59:12                 0  
21583   2011-11-22 12:48:02                 0  
29901   2010-12-07 01:59:12                 0  
32192   2011-01-31 01:07:40                 0  
...                     ...               ...  
820290  2017-12-17 01:10:52                 0  
869840  2011-09-26 00:03:49                 0  
873070  2011-09-26 00:03:49                 0  
900543  2013-08-04 16:43:08                 0  
903817  2013-08-04 16:43:08                 0  

[126 rows x 5 columns]

Tenemos 126 registros duplicados. Eliminaremos los duplicados, manteniendo la primer aparición:

In [4]:
df.drop_duplicates(keep= 'first', inplace= True)

Verificamos los duplicados nuevamente

In [5]:
ut.registros_duplicados(df)

Empty DataFrame
Columns: [user_id, business_id, text, date, compliment_count]
Index: []

Verificamos el tipo de dato de las columnas y si contienen valores nulos

In [6]:
ut.tipo_dato(df)

columna                         tipo_dato  porcentaje_nulos
0           user_id                   [<class 'str'>]               0.0
1       business_id                   [<class 'str'>]               0.0
2              text  [<class 'str'>, <class 'float'>]               0.0
3              date                   [<class 'str'>]               0.0
4  compliment_count                   [<class 'int'>]               0.0

Cambiamos el tipo de dato de la columna 'text' a string, ya que esa columna contiene comentarios de los clientes.

In [7]:
df['text'] = df['text'].astype(str)

Cambiamos el tipo de dato de la columna 'date'

In [8]:
df['date'] = pd.to_datetime(df['date'])

Cambiamos el tipo de dato de la columna 'compliment_count', esta columna se trata del número de cumplidos que ha recibido esa reseña

In [9]:
df['compliment_count'] = df['compliment_count'].astype(int)

In [10]:
df.head()

user_id             business_id  \
0  AGNUgVwnZUey3gcPCJ76iw  3uLgwr0qeCNMjKenHJwPGQ   
1  NBN4MgHP9D3cw--SnauTkA  QoezRbYQncpRqyrLH6Iqjg   
2  -copOvldyKh1qr-vzkDEvw  MYoRNLb5chwjQe3c_k37Gg   
3  FjMQVZjSqY8syIO-53KFKw  hV-bABTK-glh5wj31ps_Jw   
4  ld0AperBXk1h6UbqmM80zw  _uN0OudeJ3Zl_tf6nxg5ww   

                                                text                date  \
0                     Avengers time with the ladies. 2012-05-18 02:17:21   
1  They have lots of good deserts and tasty cuban... 2013-02-05 18:35:10   
2             It's open even when you think it isn't 2013-08-18 00:56:08   
3                          Very decent fried chicken 2017-06-27 23:05:38   
4             Appetizers.. platter special for lunch 2012-10-06 19:43:09   

   compliment_count  
0                 0  
1                 0  
2                 0  
3                 0  
4                 0

Volvemos a verificar el tipo de dato de las columnas

In [11]:
ut.tipo_dato(df)

columna                                          tipo_dato  \
0           user_id                                    [<class 'str'>]   
1       business_id                                    [<class 'str'>]   
2              text                                    [<class 'str'>]   
3              date  [<class 'pandas._libs.tslibs.timestamps.Timest...   
4  compliment_count                                    [<class 'int'>]   

   porcentaje_nulos  
0               0.0  
1               0.0  
2               0.0  
3               0.0  
4               0.0

Reseteamos el indice del DataFrame

In [13]:
df.reset_index(drop = True, inplace = True)